internal/topology/check/compatibility.go

```diff
@@ -226,20 +226,6 @@ func ClusterClassesAreCompatible(current, desired *clusterv1.ClusterClass) field
 func MachineDeploymentClassesAreCompatible(current, desired *clusterv1.ClusterClass) field.ErrorList {
 	var allErrs field.ErrorList
 
```

Comment:
Should we update Validation / ClusterClass / For object updates in the proposal accordingly?

Reply:
As discussed, let's do it in a follow-up PR to not block this PR because of proposal changes

Reply:
I think we need to take an overall look at how much implementation detail is in the proposal - let's push this out for now and I'll do an overall review.